### PR TITLE
serialize rpc inputs through 'qs' Fixes #1513

### LINKED ISSUE #1513: service blocks can't serialize input lists with null values

### PR DESCRIPTION
**Pull request by hamidzr**

### CODE REVIEW

File: test/client/debug.html [hunk 1]
```diff
@@ -0,0 +1,104 @@
+<html>
```
Comment (brollb) [inline test/client/debug.html]: why did you add this file? We shouldn't have two files for client side tests (duplicated files, that is)

**Comment by brollb** [comment]
I was initially concerned about the increase in source code size (with slow school networks this can be problematic) but it looks like the size increase is from 1133786 to 1141402 (characters). This is pretty insignificant and isn't a problem.

File: test/client/services.spec.js [hunk 2]
```diff
@@ -1,32 +1,67 @@
 /* globals expect, driver, MessageCreatorMorph, PushButtonMorph, NetsProcess*/
 describe('services', function() {
+
+
     before(function() {
         driver.selectCategory('network');
     });
 
-    describe('rpc inputs', function() {
+    describe.only('rpc inputs', function() {
```
Comment (brollb) [inline test/client/services.spec.js]: remove the `.only`

File: test/client/services.spec.js [hunk 3]
```diff
@@ -1,32 +1,67 @@
...
+            return serialized;
+        }
+
+        it('should serialize lists properly', function() {
+            var listInput = new List([new List([2,3]),new List([3,4])]);
+            var expectedOutput = 'arg[0][0]=2&arg[0][1]=3&arg[1][0]=3&arg[1][1]=4';
+            expect(serialize(listInput)).to.be(expectedOutput);
         });
 
+        it('should serialize [[0,1],[1,"null"],[2,null],[3],null,"null"]', function() {
```
Comment (brollb) [inline test/client/services.spec.js]: implement or remove this test

File: test/client/debug.html (hunk 1 above)
Comment (hamidzr) [reply test/client/debug.html]: this one serves unminified assets and makes debugging easier.

**Comment by hamidzr** [comment]
>> I was initially concerned about the increase in source code size (with slow school networks this can be problematic) but it looks like the size increase is from 1133786 to 1141402 (characters). This is pretty insignificant and isn't a problem.

so should I move debug.html over index so that all client-side tests are served with unminified files?

**Comment by brollb** [comment]
why did you change `test/client/index.html` and the `dev` service?

**Comment by hamidzr** [comment]
changed the dev service to fix the mentioned issue #1521

**Comment by hamidzr** [comment]
and I changed the index page to fix the issue #1520.

**Comment by brollb** [comment]
Separate PR's would be better for understanding why each change was made
and small PRs are easier to accept (especially when some of the changes are
a little controversial #1520).

**Comment by brollb** [comment]
Can you fix the last code climate issue (not the duplication ones)?
